src/ostorlab/scanner/handler.py

```diff
@@ -0,0 +1,229 @@
+"""Module responsible for establishing a connection with NATS."""
+
+import asyncio
+import datetime
+import logging
+import ssl
+import sys
+from typing import Optional
+
+from nats.js import errors
+import nats
+from nats.js import api as js_api
+from nats.js import client as js_client
+
+from ostorlab.scanner.proto.scan._location import startAgentScan_pb2
+
+DEFAULT_PENDING_BYTES_LIMIT = 400 * 1024 * 1024
+
+DEFAULT_CONNECT_TIMEOUT = 20
+
+DEFAULT_MAX_INFLIGHT = 1
+
+DEFAULT_ACK_WAIT = 120
+
+DEFAULT_PUBLISH_ACK_WAIT = 30
+
+logger = logging.getLogger(__name__)
+
+
+class Error(Exception):
+    """Base Error."""
+
+
+class ClientBusHandler:
+    """Handler for establishing a connection with NATS and performing client operations."""
+
+    def __init__(
+        self,
+        bus_url: str,
+        cluster_id: str,
+        name: str,
+        tls_context: Optional[ssl.SSLContext] = None,
+        loop=None,
+    ):
+        self._bus_url = bus_url
+        self._cluster_id = cluster_id
+        self._name = name
+        self._nc: nats.NATS = nats.NATS()
+        self._js: Optional[js_client.JetStreamContext] = None
+        self._loop = loop or asyncio.get_event_loop()
+        if tls_context is None and bus_url.startswith("tls://"):
+            self._tls_context = ssl.create_default_context()
+        else:
+            self._tls_context = tls_context
+
+    async def __aenter__(self) -> "ClientBusHandler":
+        await self.connect()
+        return self
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        await self.close()
+
+    async def connect(self, connect_timeout=DEFAULT_CONNECT_TIMEOUT):
+        """
+        Connect to the NATS server.
+
+        Args:
+            connect_timeout: Timeout for establishing the connection. Defaults to DEFAULT_CONNECT_TIMEOUT.
+        """
+        await self._nc.connect(
+            self._bus_url,
+            name=self._name,
+            tls=self._tls_context,
+            connect_timeout=connect_timeout,
+            error_cb=self._error_cb,
+            closed_cb=self._closed_cb,
+            reconnected_cb=self._reconnected_cb,
+        )
+        self._js = self._nc.jetstream()
+
+    async def add_stream(self, name, subjects):
+        """
+        Add a stream to the NATS server.
+
+        Args:
+            name: Name of the stream.
+            subjects: Subjects for the stream.
+        """
+        await self._js.add_stream(name=name, subjects=subjects)
+
+    async def delete_stream(self, name):
+        """
+        Delete a stream from the NATS server.
+
+        Args:
+            name: Name of the stream.
+        """
+        try:
+            await self._js.delete_stream(name=name)
+        except errors.ServiceUnavailableError as e:
```

Comment:
Are you sure this is the correct exception to catch? I think the delete raises another exception when the stream you want to delete does not exist.